File: lib_image_loader/src/main/java/com/win/lib_image_loader/image/CustomRequestListener.java
package com.win.lib_image_loader.image;

import androidx.annotation.Nullable;

import com.bumptech.glide.load.DataSource;
import com.bumptech.glide.load.engine.GlideException;
import com.bumptech.glide.request.RequestListener;
import com.bumptech.glide.request.target.Target;

public interface CustomRequestListener extends RequestListener {
    @Override
    boolean onLoadFailed(@Nullable GlideException e, Object model, Target target, boolean isFirstResource);

    @Override
    boolean onResourceReady(Object resource, Object model, Target target, DataSource dataSource, boolean isFirstResource);
}

File: lib_image_loader/src/main/java/com/win/lib_image_loader/image/CustomRequestListener.kt
package com.win.lib_image_loader.image

import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target

interface CustomRequestListener : RequestListener<Any?> {
//    override fun onLoadFailed(
//        e: GlideException?,
//        model: Any,
//        target: Target<*>?,
//        isFirstResource: Boolean
//    ): Boolean
//
//    override fun onResourceReady(
//        resource: Any?,
//        model: Any,
//        target: Target<*>?,
//        dataSource: DataSource,
//        isFirstResource: Boolean
//    ): Boolean
}